src/network/network_client.cpp

```diff
@@ -727,6 +741,15 @@ NetworkRecvStatus ClientNetworkGameSocketHandler::Receive_SERVER_ERROR(Packet *p
 
 	NetworkErrorCode error = (NetworkErrorCode)p->Recv_uint8();
 
+	/* If we query a server that is 1.11.1 or older, we get an
+	 * NETWORK_ERROR_NOT_EXPECTED on requesting the game info. Show a special
+	 * error popup in that case.
+	 */
+	if ((this->status == STATUS_GAME_INFO || this->status == STATUS_COMPANY_INFO) && error == NETWORK_ERROR_NOT_EXPECTED) {
```

Comment:
I would swap the error and status checks around. Practically the error should almost never happen, so generally it is better to put that in the begin of such a statement. Or more generally, put the cheapest and most restricting parts of an expression with && at the begin, so it does not need to evaluate as much in most of the time.
Also, mentally this branch is about the error, not so much the status. The status is just to determine whether to show it when an error occurs.

Reply:
You do realise that this event happening to start with is VERY unlikely, right? So basically you argue an optimization of nothing :D I don't mind flipping it around, but in my opinion it is the state that makes this statement special. The error is secondary :) Either side of the case, even for the function itself, is very unlikely :)